### PR TITLE
[CDAP-16105] [CDAP-16080] [CDAP-15103] validation messaging display changes

### PR DESCRIPTION
**Pull request by itsanudeep**
https://issues.cask.co/browse/CDAP-16105
Replaced banner with text in modal header per design. Same for post action plugins.

https://issues.cask.co/browse/CDAP-16080
Success message will not be shown in modal header per new design.

https://issues.cask.co/browse/CDAP-15103
Description will take as much space as it can after leaving space for buttons and validation message.

### CODE REVIEW

**Comment by itsanudeep** [comment]
RAT and Checkstyle: https://builds.cask.co/browse/CDAP-DRC9620
Build and Test: https://builds.cask.co/browse/CDAP-UDUT441

File: cdap-ui/app/directives/my-post-run-action-wizard/my-post-run-action-wizard-modal.html [hunk 1]
```diff
@@ -24,6 +24,12 @@ <h5 ng-if="action.name" class="modal-title pull-left">
       <small>{{action.description}}</small>
     </p>
   </h5>
+  <span class="text-success"
+        ng-if="errorCount === 0">No Errors found.</span>
+  <span class="text-danger"
+        ng-if="errorCount > 0">
+    {{errorCount}}
+    error{{errorCount === 1 ? '': 's'}} found.</span>
```
Comment (ajainarayanan) [inline cdap-ui/app/directives/my-post-run-action-wizard/my-post-run-action-wizard-modal.html]: Can we use ng-pluralize here?

**Comment by yukiej** [comment]
Does this PR also address the GetSchema button click resulting in validation being performed for plugin?

File: cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js [hunk 2]
```diff
@@ -450,13 +450,14 @@ class HydratorPlusPlusNodeConfigCtrl {
     return true;
   }
 
-  validatePluginProperties(callback) {
+  validatePluginProperties(callback, noSuccessIndicator) {
```
Comment (ajainarayanan) [inline cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js]: nit: don't use double negation to show the success banner. Right now for us to show the success banner we have to pass `false` to this function which is not intuitive. Can you rename the variable to be `showSuccessIndicator` and pass false to hide it?

File: cdap-ui/app/directives/my-post-run-action-wizard/my-post-run-action-wizard-modal.html [hunk 3]
```diff
@@ -24,6 +24,12 @@ <h5 ng-if="action.name" class="modal-title pull-left">
       <small>{{action.description}}</small>
     </p>
   </h5>
+  <span class="text-success"
+        ng-if="errorCount === 0">No Errors found.</span>
```
Comment (yukiej) [inline cdap-ui/app/directives/my-post-run-action-wizard/my-post-run-action-wizard-modal.html]: Nit: "errors" should be lowercase here

File: cdap-ui/app/hydrator/templates/partial/node-config-modal/popover.html [hunk 4]
```diff
@@ -26,6 +26,12 @@
       <small>{{HydratorPlusPlusNodeConfigCtrl.state.node.description}}</small>
     </p>
   </h5>
+  <span class="text-success"
+        ng-if="HydratorPlusPlusNodeConfigCtrl.errorCount === 0">No Errors found.</span>
```
Comment (yukiej) [inline cdap-ui/app/hydrator/templates/partial/node-config-modal/popover.html]: Nit: "errors" should be lowercase here

File: cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js [hunk 5]
```diff
@@ -466,6 +467,10 @@ class HydratorPlusPlusNodeConfigCtrl {
         vm.propertyErrors = {};
         vm.inputSchemaErrors = {};
         vm.outputSchemaErrors = {};
+        // Do not show success validation message for validation via get schema.
+        if (noSuccessIndicator) {
+          vm.errorCount = null;
```
Comment (yukiej) [inline cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js]: Why is errorCount being set to null here, as opposed to undefined? Might be good to update the comment on line 458 to add the null case as well.

File: cdap-ui/app/directives/my-post-run-action-wizard/my-post-run-action-wizard-modal.html (hunk 1 above)
Comment (itsanudeep) [reply cdap-ui/app/directives/my-post-run-action-wizard/my-post-run-action-wizard-modal.html]: TIL there's ng-pluralize :smile: thanks!

**Comment by itsanudeep** [comment]
> Does this PR also address the GetSchema button click resulting in validation being performed for plugin?

Yes! Validation would still happen but we will not show success-message. We'd still show failure-message if errors are found.

**Review by yukiej — APPROVED** [review]
lgtm!

**Comment by itsanudeep** [comment]
RAT and Checkstyle: https://builds.cask.co/browse/CDAP-DRC9620-3
Build and Test: https://builds.cask.co/browse/CDAP-UDUT441-3

**Review by ajainarayanan — APPROVED** [review]
👍 LGTM

**Pull request merged**